src/libraries/System.IO.MemoryMappedFiles/src/Microsoft/Win32/SafeMemoryMappedFileHandle.Unix.cs

```diff
@@ -74,7 +85,17 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
-        protected override bool ReleaseHandle() => true; // Nothing to clean up.  We unlinked immediately after creating the backing store.
+        protected override bool ReleaseHandle() {
+            Debug.Assert(!IsInvalid);
+
+            if (_fileStreamHandle != null) {
+                SetHandle((IntPtr) (-1));
+                _fileStreamHandle.DangerousRelease();
+                _fileStreamHandle = null;
+            }
+
+            return true;
+        }
```

Comment:
Can you please add test validating the desired functionality?

Reply:
Sure, I will add

Reply:
@stephentoub I've added a test to check it matches the filestream handle